### PR TITLE
Dns c tests

### PR DESCRIPTION
**Pull request by dorin-iancu**
DNS C tests + getExternalBalance fix + new myDebug function for debugging purposes (which is basically the equivalent of finish() with some printlns)

### CODE REVIEW

**Review by andrei-marinica — DISMISSED** [review]

File: arwen/elrondapi/elrondei.go [hunk 1]
```diff
@@ -455,8 +462,10 @@ func getExternalBalance(context unsafe.Pointer, addressOffset int32, resultOffse
 	}
 
 	balance := blockchain.GetBalance(address)
+	balance32Bytes := make([]byte, 32)
```
Comment (sasurobert) [inline arwen/elrondapi/elrondei.go:465]: add a constant for 32 - there should already be some constants for it.

File: arwen/elrondapi/elrondei.go [hunk 2]
```diff
@@ -1259,6 +1268,22 @@ func returnData(context unsafe.Pointer, pointer int32, length int32) {
 	output.Finish(data)
 }
 
+//export myDebug
+func myDebug(context unsafe.Pointer, pointer int32, length int32) {
+	runtime := arwen.GetRuntimeContext(context)
+	//output := arwen.GetOutputContext(context)
```
Comment (sasurobert) [inline arwen/elrondapi/elrondei.go:1274]: delete commented code

File: arwen/elrondapi/elrondei.go [hunk 3]
```diff
@@ -1259,6 +1268,22 @@ func returnData(context unsafe.Pointer, pointer int32, length int32) {
...
+	metering := arwen.GetMeteringContext(context)
+
+	gasToUse := metering.GasSchedule().ElrondAPICost.Finish
+	gasToUse += metering.GasSchedule().BaseOperationCost.PersistPerByte * uint64(length)
+	metering.UseGas(gasToUse)
+
+	data, err := runtime.MemLoad(pointer, length)
+
+	fmt.Println("Pointer:", pointer, " Length: ", length)
```
Comment (sasurobert) [inline arwen/elrondapi/elrondei.go:1283]: delete this debug function. we do not want to fill the nodes screen with these written data.

File: arwen/elrondapi/elrondei.go (hunk 1 above)
Comment (sasurobert) [inline arwen/elrondapi/elrondei.go:465]: add a constant for 32 - there should already be some constants for it.

**Review by sasurobert — CHANGES_REQUESTED** [review]

**Review by sasurobert — APPROVED** [review]

**Pull request closed without merging**